softlayer/resource_softlayer_network_vlan.go

```diff
@@ -0,0 +1,419 @@
+package softlayer
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/softlayer/softlayer-go/datatypes"
+	"github.com/softlayer/softlayer-go/filter"
+	"github.com/softlayer/softlayer-go/helpers/hardware"
+	"github.com/softlayer/softlayer-go/helpers/location"
+	"github.com/softlayer/softlayer-go/helpers/product"
+	"github.com/softlayer/softlayer-go/services"
+	"github.com/softlayer/softlayer-go/session"
+	"github.com/softlayer/softlayer-go/sl"
+	"log"
+	"time"
+)
+
+const (
+	AdditionalServicesPackageType            = "ADDITIONAL_SERVICES"
+	AdditionalServicesNetworkVlanPackageType = "ADDITIONAL_SERVICES_NETWORK_VLAN"
+)
+
+func resourceSoftLayerNetworkVlan() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceSoftLayerNetworkVlanCreate,
+		Read:     resourceSoftLayerNetworkVlanRead,
+		Update:   resourceSoftLayerNetworkVlanUpdate,
+		Delete:   resourceSoftLayerNetworkVlanDelete,
+		Exists:   resourceSoftLayerNetworkVlanExists,
+		Importer: &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"id": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"datacenter": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"type": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"primary_subnet_size": &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"primary_router_hostname": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
```

Comment:
Actually, I couldn't find other examples which use both Computed and Optional, but here is a test result of `Optional` property only.

_vlan.tf_
```
resource "softlayer_network_vlan" "test_vlan" {
   name = "test_vlan2"
   datacenter = "lon02"
   type = "PUBLIC"
   primary_subnet_size = 8
}
```
 
_terraform_ _show_ (after _terraform_ _apply_)
```
softlayer_network_vlan.test_vlan:
  id = 1369721
  child_resource_count = 0
  datacenter = lon02
  name = test_vlan2
  primary_router_hostname = fcr01a.lon02
  primary_subnet_size = 8
  softlayer_managed = false
  subnets.# = 1
  subnets.1384167489.subnet = 159.8.147.208/29
  subnets.1384167489.subnet_type = ADDITIONAL_PRIMARY
  type = PUBLIC
  vlan_number = 1255
```

_terraform_ _plan_
```
softlayer_network_vlan.test_vlan
primary_router_hostname: "fcr01a.lon02" => ""
```

With current codes, terraform tries to delete primary_router_hostname whenever I execute `terraform apply`